### PR TITLE
convert to ES modules

### PR DESCRIPTION
**Pull request by connorjclark**
Lighthouse 10.0 will be ES modules (#12689) which will require plugins to be ES modules too. Here's a PR that updates everything in this repo to ES modules.

### CODE REVIEW

**Comment by connorjclark** [comment]
re: failures

- I get the same smoke failure on main.
- LHCI doesn't support es modules, yet.

**Comment by connorjclark** [comment]
Confirmed working with https://github.com/GoogleChrome/lighthouse/pull/14241

File: lighthouse-plugin-publisher-ads/audits/ad-blocking-tasks.js [hunk 1]
```diff
@@ -70,7 +71,7 @@ const HEADINGS = [
 class AdBlockingTasks extends Audit {
   /**
    * @return {LH.Audit.Meta}
-   * @override
+   * /override This member cannot have a JSDoc comment with an '@override' tag because its containing class ... does not extend another class.
```
Comment (jburger424) [inline lighthouse-plugin-publisher-ads/audits/ad-blocking-tasks.js:74]: Can we remove these comments?

File: lighthouserc.cjs [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// LHCI currently does not support .cjs, so generate a json config with:
```
Comment (jburger424) [inline lighthouserc.cjs:1]: Any reason for keeping this file if unsupported? Any issue with just keeping the json file?

File: package.json [hunk 3]
```diff
@@ -21,13 +22,15 @@
   "dependencies": {
     "@tusbar/cache-control": "^0.3.1",
     "@types/esprima": "^4.0.2",
-    "lighthouse": "^8.6.0",
+    "lighthouse": "next",
```
Comment (jburger424) [inline package.json:25]: I worry that if there are breaking changes they will no longer be version tied, and older versions will fail to work. Does this seem like a reasonable concern?

**Comment by jburger424** [comment]
> re: failures
> 
> * I get the same smoke failure on main.
> * LHCI doesn't support es modules, yet.

The smoke test seems to be flaky, trying to see if rerunning works. But I've gone ahead and temporarily remove LHCI from the required actions list.

File: lighthouse-plugin-publisher-ads/audits/ad-blocking-tasks.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-plugin-publisher-ads/audits/ad-blocking-tasks.js:74]: Yeah, I'll remove the entire line.

File: lighthouserc.cjs (hunk 2 above)
Comment (connorjclark) [reply lighthouserc.cjs:1]: just that .cjs is easier to author, and it is on my backlog to update LHCI to support ESM config/at least support .cjs and then I'll come back and delete the JSON here.

I can do as you suggest if you'd prefer.

File: package.json (hunk 3 above)
Comment (connorjclark) [reply package.json:25]: I'll change this to a specific nightly version. And modify again when 10.0 is actually released, ~Sept.

If you'd rather not cut an actual release to npm, that'd be fine. We can always tie Lighthouse's version of pubads to a branch or specific commit during the transition.

File: lighthouse-plugin-publisher-ads/audits/ad-blocking-tasks.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-plugin-publisher-ads/audits/ad-blocking-tasks.js]: BTW I'm not sure why this error occurs now, but it didn't seem worth the effort to figure it out.

**Review by jburger424 — APPROVED** [review]

**Comment by jburger424** [comment]
LGTM, I think best to wait for a stable LH dependency before cutting a new prod release.

**Pull request merged**